text/data_versioning.md

```diff
@@ -0,0 +1,153 @@
+
+### GATK RNA-Seq short variant discovery best practices workflow
+
+The GATK best practices workflows are de facto standards for data processing in bioinformatics. Therefore we will reuse their existing workflow for generating variant calls from RNA-Seq data. The workflow accepts a STAR-aligned BAM file, such as those produced by our RNA-Seq alignment harmonization workflow, and produces a set of variant calls by running a number of GATK steps.
+
+![GATK RNA-Seq short variant discovery workflow diagram](../resources/data_versioning/rnaseq-variant-workflow.png)
+
+#### Workflow
+
+1. Run Picard `MarkDuplicates`
+
+    ```bash
+        picard -Xmx~{java_heap_size}g MarkDuplicates \
+            -I ~{bam} \
+            -O ~{if create_bam then prefix + ".bam" else "/dev/null"} \
+            --VALIDATION_STRINGENCY SILENT \
+            --CREATE_INDEX ~{create_bam} \
+            --CREATE_MD5_FILE ~{create_bam} \
+            --COMPRESSION_LEVEL 5 \
+            --METRICS_FILE ~{prefix}.metrics.txt
+    ```
+
+2. Run GATK `SplitNCigarReads`
+
+    ```bash
+        gatk \
+            SplitNCigarReads \
+            -R ~{fasta} \
+            -I ~{bam} \
+            -O ~{prefix}.bam \
+            -OBM true
+       # GATK is unreasonable and uses the plain ".bai" suffix.
+       mv ~{prefix}.bai ~{prefix}.bam.bai
+    ```
+
+3. Run GATK Spark-enabled `BaseRecalibrator` (`BaseRecalibratorSpark`).
+
+   ```bash
+        gatk \
+            --java-options "-Xms4000m" \
+            BaseRecalibratorSpark \
+            -R ~{fasta} \
+            -I ~{bam} \
+            ~{if use_original_quality_scores then "--use-original-qualities" else "" } \
+            -O ~{prefix}.txt \
+            -known-sites ~{dbSNP_vcf} \
+            -known-sites ~{sep=" --known-sites " known_indels_sites_VCFs} \
+            --spark-master local[4]
+   ```
+
+4. Run GATK Spark-enabled `ApplyBQSR` (`ApplyBQSRSpark`).
+
+    ```bash
+            gatk \
+            --java-options "-Xms3000m" \
+            ApplyBQSRSpark \
+            --spark-master local[4] \
+            -I ~{bam} \
+            ~{if use_original_quality_scores then "--use-original-qualities" else "" } \
+            -O ~{prefix}.bam \
+            --bqsr-recal-file ~{recalibration_report}
+    ```
+
+5. Run Picard `IntervalListTools` to generate list of regions to scatter over.
+
+    ```bash
+        picard -Xms1g \
+            IntervalListTools \
+            SCATTER_COUNT=~{scatter_count} \
+            SUBDIVISION_MODE=BALANCING_WITHOUT_INTERVAL_SUBDIVISION_WITH_OVERFLOW \
+            UNIQUE=~{unique} \
+            SORT=~{sort} \
+            INPUT=~{interval_list} \
+            OUTPUT=out
+    ```
+
+6. Run GATK Spark-enabled `HaplotypeCaller` (`HaplotypeCallerSpark`) on each region.
+
+    ```bash
+        gatk \
+           --java-options "-Xms6000m" \
+            HaplotypeCallerSpark \
```

Comment:
Okay so, in particular, the proposal is focused on _germline_ mutations? Is there any gap with respect to and/or benefit to calling somatic mutations?

Reply:
Yes, this is focused on `germline` mutations. The goal is the capture alterations to the pipeline that change a meaningful result, in this case, variant calls. I would expect any alignment-related issues that would affect `somatic` variant calling would also affect `germline` variant calling.